### ISSUE TITLE
Add JSON schemas to JSON store

### ISSUE THREAD

**Issue by bdarcus** [citation-style-language/schema#279]
Shouldn't we add our schemas to here, maybe when we release 1.1?

https://www.schemastore.org/json/

**Comment by denismaier** [comment]
We should. Would be even useful for yaml validation in vscode

**Comment by bdarcus** [comment]
> We should. Would be even useful for yaml validation in vscode.

Right; the [YAML LSP](https://github.com/redhat-developer/yaml-language-server#language-server-settings) will do validation using these schemas.

Let's do this.

@dhimmel - any thoughts on the versioning question here? Should we just do latest/master, at least to start?

Hold off until 1.1 is merged?

**Comment by bdarcus** [comment]
Here's the config content:

``` js
{
  "name": "bower.json",
  "description": "Bower package description file",
  "fileMatch": [ "bower.json", ".bower.json" ],
  "url": "http://schemastore.org/schemas/json/bower"
}
```

Is there any room for us defining new extensions here>

If not, how would we specify file matching?

**Comment by dhimmel** [comment]
> Would be even useful for yaml validation in vscode

So this would allow vscode to autocomplete and validate CSL JSON.YAML? How does a text editor know that any given `.json` or `.yaml` file is CSL JSON and to use this spec? Is there something the file needs to include or does the user need to tell the text editor that the file should follow a certain schema?

**Comment by bdarcus** [comment]
In Atom, you can specify based on file extensions, etc, or allow the user to manually choose the grammar.

Not sure here.

**Comment by denismaier** [comment]
Glob patterns work in vscode

**Comment by dhimmel** [comment]
Ideally there would be a way for the JSON file to specify what schema it intends to follow. VS Code [supports](https://code.visualstudio.com/docs/languages/json#_mapping-in-the-json) `$schema`, but that is not a standard and I'm not sure it'd work for CSL JSON which uses an array as the top-level.

VS Code also supports a [workspace setting](https://code.visualstudio.com/docs/languages/json#_mapping-to-a-schema-in-the-workspace), which is what I think @denismaier mentions in "Glob patterns work in vscode".

I have been naming my CSL JSON files like `references.json`. Do people here tend to use more verbose extensions to indicate CSL JSON? Like `references.csl.json` or `references-csl.json`? That would allow for a precise glob.

> @dhimmel - any thoughts on the versioning question here? Should we just do latest/master, at least to start?

not yet

**Comment by bdarcus** [comment]
> I have been naming my CSL JSON files like `references.json`. Do people here tend to use more verbose extensions to indicate CSL JSON? Like `references.csl.json` or `references-csl.json`? That would allow for a precise glob.

Not sure, but we could add it and recommend? 

So `*.csl.json` and `*.csl.yaml`?

> not yet

Let's sort out the outstanding issues with the schema for 1.1, and then we can do this then?

**Comment by dhimmel** [comment]
I added the following in my workspace's `.vscode/settings.json` and the auto-complete is now working!

```json
    "json.schemas": [
        {
            "fileMatch": [
                "*references*.json"
            ],
            "url": "https://github.com/dhimmel/csl-schema/raw/manubot/csl-data.json"
        }
    ]
```

So users can always specify https://resource.citationstyles.org/schema/latest/input/json/csl-data.json for the schema, but does JSON Store make this easier?

**Comment by bdarcus** [comment]
I know some tools have support for the store, so you can probably select the schema without knowing the URL?

That was the impression I got.
